src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2473,21 +2495,23 @@ public bool HasBoxingConversion(TypeSymbol source, TypeSymbol destination, ref H
             if ((source.TypeKind == TypeKind.TypeParameter) &&
                 HasImplicitBoxingTypeParameterConversion((TypeParameterSymbol)source, destination, ref useSiteDiagnostics))
             {
-                return true;
+                return ConversionKind.TypeParameterBoxing;
```

Comment:
This seems trivially true.  If source is a type parameter, and we have an implicit boxing type parameter conversion, then this is a TypeParameterBoxing conversion.

The quesiton then remains what happens after this point.  It's still possible to for source ot be a type parameter, but is to not have an *implcit* type parameter boxing conversion.  Are there still cases where we would want to say this is type-parameter boxing?  Or should all the remainder be "value type boxing".  

I've chosen "value type boxing" for everything past this point, as the only boxing we get is because the type parameter is known to be a value-type (i.e. it has the struct constraint).  Because of that, it's really the inherit value-type boxing that is happening, and it's not really relevant if that value type was a type parameter (IMO at least).

Let me know what you think.